diff -Nrup A/system_stm32f30x.c B/system_stm32f30x.c
--- A/system_stm32f30x.c	2013-01-21 19:27:28.000000000 +0100
+++ B/system_stm32f30x.c	2019-12-03 20:23:58.579208083 +0100
@@ -111,6 +111,7 @@
   */
 
 #include "stm32f30x.h"
+#include "config.h" /* CooCox workaround */
 
 /**
   * @}
@@ -130,8 +131,12 @@
 /*!< Uncomment the following line if you need to relocate your vector Table in
      Internal SRAM. */ 
 /* #define VECT_TAB_SRAM */
+#if defined(Bootloader)
+#define VECT_TAB_OFFSET  0x3000
+#else
 #define VECT_TAB_OFFSET  0x0 /*!< Vector Table base offset field. 
                                   This value must be a multiple of 0x200. */  
+#endif
 /**
   * @}
   */ 
