--- family.c.orig	2024-11-28 10:41:15.429260889 +0100
+++ family.c	2025-01-17 21:34:28.409496327 +0100
@@ -79,7 +79,12 @@ bool __no_inline_not_in_flash_func(get_b
 
   // The HI GPIO registers in SIO can observe and control the 6 QSPI pins.
   // Note the button pulls the pin *low* when pressed.
-  bool button_state = (sio_hw->gpio_hi_in & (1u << CS_PIN_INDEX));
+  #if PICO_RP2040
+    #define CS_BIT (1u << 1)
+  #else
+    #define CS_BIT SIO_GPIO_HI_IN_QSPI_CSN_BITS
+  #endif
+  bool button_state = (sio_hw->gpio_hi_in & CS_BIT);
 
   // Need to restore the state of chip select, else we are going to have a
   // bad time when we return to code in flash!
