diff -Nrup A/startup/startup_stm32f10x_md.s B/startup/startup_stm32f10x_md.s
--- A/startup/startup_stm32f10x_hd.s	2017-05-05 16:10:18.000000000 +0200
+++ B/startup/startup_stm32f10x_hd.s	2020-04-30 03:52:33.915828904 +0200
@@ -102,7 +102,7 @@ LoopFillZerobss:
 /* Call the clock system intitialization function.*/
     bl  SystemInit
 /* Call static constructors */
-    bl __libc_init_array
+//    bl __libc_init_array
 /* Call the application's entry point.*/
 	bl	main
 	bx	lr
@@ -134,7 +134,7 @@ Infinite_Loop:
 
 
 g_pfnVectors:
-	.word	_estack
+	.word	_eram
 	.word	Reset_Handler
 	.word	NMI_Handler
 	.word	HardFault_Handler
