diff -Nrup A/stm32f10x_conf.h B/stm32f10x_conf.h
--- A/stm32f10x_conf.h	2017-06-16 10:14:04.000000000 +0200
+++ B/stm32f10x_conf.h	2020-04-30 02:58:38.655638934 +0200
@@ -42,6 +42,7 @@
 
 /* Includes ------------------------------------------------------------------*/
 /* Uncomment the line below to enable peripheral header file inclusion */
+#if 0
 #include "stm32f10x_adc.h"
 #include "stm32f10x_bkp.h"
 #include "stm32f10x_can.h"
@@ -64,6 +65,16 @@
 #include "stm32f10x_usart.h"
 #include "stm32f10x_wwdg.h"
 #include "misc.h"   /* High level functions for NVIC and SysTick (add-on to CMSIS functions) */
+#endif
+#include "stm32f10x_exti.h"
+#include "stm32f10x_flash.h"
+#include "stm32f10x_gpio.h"
+#include "stm32f10x_pwr.h"
+#include "stm32f10x_rcc.h"
+#include "stm32f10x_tim.h"
+#include "stm32f10x_usart.h"
+#include "misc.h"
+#include "config.h"
 
 /* Exported types ------------------------------------------------------------*/
 /* Exported constants --------------------------------------------------------*/
