diff -Nrup a/system_stm32f10x.c b/system_stm32f10x.c
--- a/system_stm32f10x.c	2012-03-22 15:44:16.000000000 +0100
+++ b/system_stm32f10x.c	2017-02-14 16:59:30.667474139 +0100
@@ -69,6 +69,7 @@
   */
 
 #include "stm32f10x.h"
+#include "config.h" /* CooCox workaround */
 
 /**
   * @}
@@ -131,8 +132,12 @@
 /*!< Uncomment the following line if you need to relocate your vector Table in
      Internal SRAM. */ 
 /* #define VECT_TAB_SRAM */
+#if defined(Bootloader)
+#define VECT_TAB_OFFSET  0x5000
+#else
 #define VECT_TAB_OFFSET  0x0 /*!< Vector Table base offset field. 
                                   This value must be a multiple of 0x200. */
+#endif
 
 
 /**
