diff -Nrup a/inc/eeprom.h b/inc/eeprom.h
--- a/inc/eeprom.h	2009-08-04 17:25:52.000000000 +0200
+++ b/inc/eeprom.h	2014-11-19 14:58:32.655027900 +0100
@@ -35,8 +35,8 @@
 #endif
 
 /* EEPROM start address in Flash */
-#define EEPROM_START_ADDRESS    ((uint32_t)0x08010000) /* EEPROM emulation start address:
-                                                  after 64KByte of used Flash memory */
+#define EEPROM_START_ADDRESS    ((uint32_t)0x0801F800) /* F103C8 has actually 128k = 0x20000 Flash */
+                                                       /* 0x8020000 - 2 * 0x400 = 0x801F800 */
 
 /* Pages 0 and 1 base and end addresses */
 #define PAGE0_BASE_ADDRESS      ((uint32_t)(EEPROM_START_ADDRESS + 0x000))
@@ -65,7 +65,7 @@
 #define PAGE_FULL               ((uint8_t)0x80)
 
 /* Variables' number */
-#define NumbOfVar               ((uint8_t)0x03)
+#define NumbOfVar               ((uint8_t)0x0F)
 
 /* Exported types ------------------------------------------------------------*/
 /* Exported macro ------------------------------------------------------------*/
