diff -Nrup a/stm32f10x_conf.h b/stm32f10x_conf.h
--- a/stm32f10x_conf.h	2013-01-21 19:27:28.000000000 +0100
+++ b/stm32f10x_conf.h	2014-12-13 00:02:07.000000000 +0100
@@ -32,6 +32,7 @@
 
 /* Includes ------------------------------------------------------------------*/
 /* Uncomment the line below to enable peripheral header file inclusion */
+#if 0
 #include "stm32f10x_adc.h"
 #include "stm32f10x_bkp.h"
 #include "stm32f10x_can.h"
@@ -54,6 +55,16 @@
 #include "stm32f10x_usart.h"
 #include "stm32f10x_wwdg.h"
 #include "misc.h"   /* High level functions for NVIC and SysTick (add-on to CMSIS functions) */
+#endif
+#include "stm32f10x_exti.h"
+#include "stm32f10x_flash.h"
+#include "stm32f10x_gpio.h"
+#include "stm32f10x_pwr.h"
+#include "stm32f10x_rcc.h"
+#include "stm32f10x_tim.h"
+#include "stm32f10x_usart.h"
+#include "misc.h"
+#include "config.h"
 
 /* Exported types ------------------------------------------------------------*/
 /* Exported constants --------------------------------------------------------*/
