From: Manuel Reimer <manuel.reimer@gmx.de>
Subject: Fix to make the USB_HID code build with "-Os" using gcc 4.9

diff -Nrup a/src/usb_regs.c b/src/usb_regs.c
--- a/src/usb_regs.c	2013-01-21 19:27:32.000000000 +0100
+++ b/src/usb_regs.c	2015-03-30 22:21:36.886935857 +0200
@@ -476,7 +476,7 @@ uint8_t GetEPAddress(uint8_t bEpNum)
 * Output         : None.
 * Return         : None.
 *******************************************************************************/
-void SetEPTxAddr(uint8_t bEpNum, uint16_t wAddr)
+void SetEPTxAddr(uint8_t bEpNum, volatile uint16_t wAddr)
 {
   _SetEPTxAddr(bEpNum, wAddr);
 }
@@ -488,7 +488,7 @@ void SetEPTxAddr(uint8_t bEpNum, uint16_
 * Output         : None.
 * Return         : None.
 *******************************************************************************/
-void SetEPRxAddr(uint8_t bEpNum, uint16_t wAddr)
+void SetEPRxAddr(uint8_t bEpNum, volatile uint16_t wAddr)
 {
   _SetEPRxAddr(bEpNum, wAddr);
 }
