diff -Nru maple-bootloader-unify_platforms.orig/Makefile maple-bootloader-unify_platforms.red.3k/Makefile
--- maple-bootloader-unify_platforms.orig/Makefile	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/Makefile	2015-02-21 19:39:13.183322933 +0100
@@ -17,7 +17,7 @@
 ST_USB = usb_lib
 
 # Optimization level [0,1,2,3,s]
-OPT ?= 0
+OPT ?= 3
 DEBUG = 
 #DEBUG = dwarf-2
 
diff -Nru maple-bootloader-unify_platforms.orig/common.h maple-bootloader-unify_platforms.red.3k/common.h
--- maple-bootloader-unify_platforms.orig/common.h	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/common.h	2015-01-13 03:54:10.000000000 +0100
@@ -38,6 +38,7 @@
 #include "stm32f10x_type.h"
 #include "cortexm3_macro.h"
 #include "usb.h"
+#include "dfu.h"
 
 typedef void (*FuncPtr)(void);
 
diff -Nru maple-bootloader-unify_platforms.orig/config.h maple-bootloader-unify_platforms.red.3k/config.h
--- maple-bootloader-unify_platforms.orig/config.h	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/config.h	2015-02-21 19:39:20.637322502 +0100
@@ -37,15 +37,19 @@
 
 # define LED_BANK GPIOA
 # define LED_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOA
-# define LED      5
+# define LED      9 // toggle LED
+# define LED2_BANK GPIOB
+# define LED2_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOB
+# define LED2      13 // toggle LED
+# define REMAP
 
 # define BUTTON_BANK GPIOC
 # define BUTTON_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOC
 # define BUTTON      9
 
-# define USB_DISC_BANK GPIOC
-# define USB_DISC_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOC
-# define USB_DISC      12
+# define USB_DISC_BANK GPIOA
+# define USB_DISC_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOA
+# define USB_DISC      13 // most easy to solder on red ST-Link
 
 #elif defined MAPLE_MINI
 
@@ -73,7 +77,7 @@
 
 #define USER_CODE_RAM     ((u32)0x20000C00)
 #define RAM_END           ((u32)0x20005000)
-#define USER_CODE_FLASH   ((u32)0x08005000)
+#define USER_CODE_FLASH   ((u32)0x08003000)
 #define FLASH_END         ((u32)0x08020000)
 
 #define VEND_ID0 0xAF
diff -Nru maple-bootloader-unify_platforms.orig/dfu.c maple-bootloader-unify_platforms.red.3k/dfu.c
--- maple-bootloader-unify_platforms.orig/dfu.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/dfu.c	2015-01-20 02:23:07.000000000 +0100
@@ -398,8 +398,10 @@
             if (code_copy_lock == BEGINNING) {
                 code_copy_lock = MIDDLE;
                 strobePin(LED_BANK, LED, 2, 0x1000);
+                strobePin(LED2_BANK, LED2, 2, 0x1000);                
                 dfuCopyBufferToExec();
                 strobePin(LED_BANK, LED, 2, 0x500);
+                strobePin(LED2_BANK, LED2, 2, 0x500);                
                 code_copy_lock = END;
             }
         }
diff -Nru maple-bootloader-unify_platforms.orig/hardware.c maple-bootloader-unify_platforms.red.3k/hardware.c
--- maple-bootloader-unify_platforms.orig/hardware.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/hardware.c	2015-01-21 15:33:29.000000000 +0100
@@ -101,7 +101,24 @@
     /* Setup APB2 for LED GPIO bank */
     rwmVal =  GET_REG(RCC_APB2ENR);
     rwmVal |= LED_RCC_APB2ENR_GPIO;
+#ifdef REMAP    
+    rwmVal |= RCC_APB2ENR_AFIOEN; /* enable AFIO clock */
+#endif    
     SET_REG(RCC_APB2ENR, rwmVal);
+    
+#ifdef REMAP
+    /* disable SWJ */
+    rwmVal =  GET_REG(AFIO_MAPR);
+    rwmVal &= ~AFIO_MAPR_SWJ_CFG;
+    rwmVal |= AFIO_MAPR_SWJ_CFG_DISABLE;
+    SET_REG(AFIO_MAPR, rwmVal);
+#endif        
+
+    /* Setup APB2 for LED2 GPIO bank */
+    rwmVal =  GET_REG(RCC_APB2ENR);
+    rwmVal |= LED2_RCC_APB2ENR_GPIO;
+    SET_REG(RCC_APB2ENR, rwmVal);
+
 
 #if (LED < 8)
 # define LED_GPIO_CR GPIO_CRL(LED_BANK)
@@ -111,6 +128,14 @@
 # define LED_CR_PORT (LED - 8)
 #endif
 
+#if (LED2 < 8)
+# define LED2_GPIO_CR GPIO_CRL(LED2_BANK)
+# define LED2_CR_PORT (LED2)
+#else
+# define LED2_GPIO_CR GPIO_CRH(LED2_BANK)
+# define LED2_CR_PORT (LED2 - 8)
+#endif
+
     /* Setup GPIO Pin as PP Out */
     rwmVal =  GET_REG(LED_GPIO_CR);
     rwmVal &= ~(0xF << (LED_CR_PORT * 4));
@@ -118,6 +143,14 @@
     SET_REG(LED_GPIO_CR, rwmVal);
 
     setPin(LED_BANK, LED);
+    
+    /* Setup GPIO Pin as PP Out */
+    rwmVal =  GET_REG(LED2_GPIO_CR);
+    rwmVal &= ~(0xF << (LED2_CR_PORT * 4));
+    rwmVal |= (0x1 << (LED2_CR_PORT * 4));
+    SET_REG(LED2_GPIO_CR, rwmVal);
+
+    setPin(LED2_BANK, LED2);    
 }
 
 void setupBUTTON(void) {
@@ -177,7 +210,7 @@
     flashLock();
     usbDsbISR();
     nvicDisableInterrupts();
-    setPin(USB_DISC_BANK, USB_DISC); // disconnect usb from host
+    resetPin(USB_DISC_BANK, USB_DISC); // disconnect usb from host
     systemReset(); // resets clocks and periphs, not core regs
 
 
diff -Nru maple-bootloader-unify_platforms.orig/hardware.h maple-bootloader-unify_platforms.red.3k/hardware.h
--- maple-bootloader-unify_platforms.orig/hardware.h	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/hardware.h	2015-01-17 04:03:02.000000000 +0100
@@ -35,6 +35,10 @@
 #define GPIOA ((u32)0x40010800)
 #define GPIOB ((u32)0x40010C00)
 #define GPIOC ((u32)0x40011000)
+#define AFIO  ((u32)0x40010000)
+#define AFIO_MAPR (AFIO + 0x04)
+#define AFIO_MAPR_SWJ_CFG (0x7<<24)
+#define AFIO_MAPR_SWJ_CFG_DISABLE (0x4<<24)
 
 #define RCC_CR      RCC
 #define RCC_CFGR    (RCC + 0x04)
@@ -46,6 +50,7 @@
 #define RCC_APB2ENR_GPIOB 0x00000008
 #define RCC_APB2ENR_GPIOC 0x00000010
 #define RCC_APB1ENR_USB 0x00800000
+#define RCC_APB2ENR_AFIOEN 0x00000001
 
 #define FLASH_ACR     (FLASH + 0x00)
 #define FLASH_KEYR    (FLASH + 0x04)
diff -Nru maple-bootloader-unify_platforms.orig/main.c maple-bootloader-unify_platforms.red.3k/main.c
--- maple-bootloader-unify_platforms.orig/main.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/main.c	2015-01-20 03:13:58.000000000 +0100
@@ -38,19 +38,20 @@
     setupCLK();
     setupLED();
     setupUSB();
-    setupBUTTON();
     setupFLASH();
 
     strobePin(LED_BANK, LED, STARTUP_BLINKS, BLINK_FAST);
+    strobePin(LED2_BANK, LED2, STARTUP_BLINKS, BLINK_FAST);    
 
     /* wait for host to upload program or halt bootloader */
-    bool no_user_jump = !checkUserCode(USER_CODE_FLASH) && !checkUserCode(USER_CODE_RAM) || readPin(BUTTON_BANK, BUTTON);
+    bool no_user_jump = !checkUserCode(USER_CODE_FLASH) && !checkUserCode(USER_CODE_RAM);
     int delay_count = 0;
 
     while ((delay_count++ < BOOTLOADER_WAIT)
             || no_user_jump) {
 
         strobePin(LED_BANK, LED, 1, BLINK_SLOW);
+        strobePin(LED2_BANK, LED2, 1, BLINK_SLOW);        
 
         if (dfuUploadStarted()) {
             dfuFinishUpload(); // systemHardReset from DFU once done
@@ -64,6 +65,7 @@
     } else {
         // some sort of fault occurred, hard reset
         strobePin(LED_BANK, LED, 5, BLINK_FAST);
+        strobePin(LED2_BANK, LED2, 5, BLINK_FAST);        
         systemHardReset();
     }
 
diff -Nru maple-bootloader-unify_platforms.orig/usb.c maple-bootloader-unify_platforms.red.3k/usb.c
--- maple-bootloader-unify_platforms.orig/usb.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/usb.c	2015-01-17 04:40:37.000000000 +0100
@@ -63,6 +63,8 @@
 
     /* initialize the usb application */
     resetPin(USB_DISC_BANK, USB_DISC);  /* present ourselves to the host */
+    strobePin(LED_BANK, LED, 1, BLINK_SLOW); // pull USBDP down for 1 blink
+    setPin(USB_DISC_BANK, USB_DISC);
     usbAppInit();
 }
 
diff -Nru maple-bootloader-unify_platforms.orig/usb_descriptor.c maple-bootloader-unify_platforms.red.3k/usb_descriptor.c
--- maple-bootloader-unify_platforms.orig/usb_descriptor.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.red.3k/usb_descriptor.c	2015-02-21 19:39:29.610321984 +0100
@@ -176,7 +176,7 @@
     0x03,
     'D', 0, 'F', 0, 'U', 0, ' ', 0, 'P', 0, 'r', 0, 'o', 0, 'g', 0, 'r', 0,
     'a', 0, 'm', 0, ' ', 0, 'F', 0, 'L', 0, 'A', 0, 'S', 0, 'H', 0, ' ', 0,
-    '0', 0, 'x', 0, '0', 0, '8', 0, '0', 0, '0', 0, '5', 0, '0', 0, '0', 0,
+    '0', 0, 'x', 0, '0', 0, '8', 0, '0', 0, '0', 0, '3', 0, '0', 0, '0', 0,
     '0', 0
 };
 
