diff -Nrup maple-bootloader-unify_platforms.orig/Makefile maple-bootloader-unify_platforms.dev/Makefile
--- maple-bootloader-unify_platforms.orig/Makefile	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/Makefile	2015-02-21 16:33:33.875965973 +0100
@@ -17,7 +17,7 @@ ST_LIB = stm32_lib
 ST_USB = usb_lib
 
 # Optimization level [0,1,2,3,s]
-OPT ?= 0
+OPT ?= s
 DEBUG = 
 #DEBUG = dwarf-2
 
diff -Nrup maple-bootloader-unify_platforms.orig/common.h maple-bootloader-unify_platforms.dev/common.h
--- maple-bootloader-unify_platforms.orig/common.h	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/common.h	2015-01-19 22:12:28.400958880 +0100
@@ -38,6 +38,7 @@
 #include "stm32f10x_type.h"
 #include "cortexm3_macro.h"
 #include "usb.h"
+#include "dfu.h"
 
 typedef void (*FuncPtr)(void);
 
diff -Nrup maple-bootloader-unify_platforms.orig/config.h maple-bootloader-unify_platforms.dev/config.h
--- maple-bootloader-unify_platforms.orig/config.h	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/config.h	2015-02-26 16:57:37.309130543 +0100
@@ -35,17 +35,17 @@
 
 #ifdef MAPLE
 
-# define LED_BANK GPIOA
-# define LED_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOA
-# define LED      5
+# define LED_BANK GPIOB
+# define LED_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOB
+# define LED      13 // toggle LED
 
 # define BUTTON_BANK GPIOC
 # define BUTTON_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOC
 # define BUTTON      9
 
-# define USB_DISC_BANK GPIOC
-# define USB_DISC_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOC
-# define USB_DISC      12
+# define USB_DISC_BANK GPIOB
+# define USB_DISC_RCC_APB2ENR_GPIO RCC_APB2ENR_GPIOB
+# define USB_DISC      15
 
 #elif defined MAPLE_MINI
 
@@ -73,7 +73,7 @@
 
 #define USER_CODE_RAM     ((u32)0x20000C00)
 #define RAM_END           ((u32)0x20005000)
-#define USER_CODE_FLASH   ((u32)0x08005000)
+#define USER_CODE_FLASH   ((u32)0x08002000)
 #define FLASH_END         ((u32)0x08020000)
 
 #define VEND_ID0 0xAF
diff -Nrup maple-bootloader-unify_platforms.orig/hardware.c maple-bootloader-unify_platforms.dev/hardware.c
--- maple-bootloader-unify_platforms.orig/hardware.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/hardware.c	2015-01-19 22:12:28.400958880 +0100
@@ -177,7 +177,7 @@ void jumpToUser(u32 usrAddr) {
     flashLock();
     usbDsbISR();
     nvicDisableInterrupts();
-    setPin(USB_DISC_BANK, USB_DISC); // disconnect usb from host
+    resetPin(USB_DISC_BANK, USB_DISC); // disconnect usb from host
     systemReset(); // resets clocks and periphs, not core regs
 
 
diff -Nrup maple-bootloader-unify_platforms.orig/main.c maple-bootloader-unify_platforms.dev/main.c
--- maple-bootloader-unify_platforms.orig/main.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/main.c	2015-01-19 22:18:42.849937264 +0100
@@ -38,13 +38,13 @@ int main() {
     setupCLK();
     setupLED();
     setupUSB();
-    setupBUTTON();
+    //setupBUTTON();
     setupFLASH();
 
     strobePin(LED_BANK, LED, STARTUP_BLINKS, BLINK_FAST);
 
     /* wait for host to upload program or halt bootloader */
-    bool no_user_jump = !checkUserCode(USER_CODE_FLASH) && !checkUserCode(USER_CODE_RAM) || readPin(BUTTON_BANK, BUTTON);
+    bool no_user_jump = !checkUserCode(USER_CODE_FLASH) && !checkUserCode(USER_CODE_RAM);
     int delay_count = 0;
 
     while ((delay_count++ < BOOTLOADER_WAIT)
diff -Nrup maple-bootloader-unify_platforms.orig/usb.c maple-bootloader-unify_platforms.dev/usb.c
--- maple-bootloader-unify_platforms.orig/usb.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/usb.c	2015-02-26 17:43:15.589291983 +0100
@@ -50,7 +50,7 @@ void setupUSB(void) {
 #endif
 
     /* preset pin to high */
-    setPin(USB_DISC_BANK, USB_DISC);
+    //setPin(USB_DISC_BANK, USB_DISC);
 
     /* Setup GPIO Pin as OD out */
     rwmVal  = GET_REG(USB_DISC_GPIO_CR);
@@ -62,7 +62,7 @@ void setupUSB(void) {
     pRCC->APB1ENR |= RCC_APB1ENR_USB;
 
     /* initialize the usb application */
-    resetPin(USB_DISC_BANK, USB_DISC);  /* present ourselves to the host */
+    setPin(USB_DISC_BANK, USB_DISC);  /* present ourselves to the host */
     usbAppInit();
 }
 
diff -Nrup maple-bootloader-unify_platforms.orig/usb_descriptor.c maple-bootloader-unify_platforms.dev/usb_descriptor.c
--- maple-bootloader-unify_platforms.orig/usb_descriptor.c	2014-05-14 15:45:50.000000000 +0200
+++ maple-bootloader-unify_platforms.dev/usb_descriptor.c	2015-02-21 13:53:49.059519276 +0100
@@ -176,7 +176,7 @@ u8 u8_usbStringAlt1[0x3A] = {
     0x03,
     'D', 0, 'F', 0, 'U', 0, ' ', 0, 'P', 0, 'r', 0, 'o', 0, 'g', 0, 'r', 0,
     'a', 0, 'm', 0, ' ', 0, 'F', 0, 'L', 0, 'A', 0, 'S', 0, 'H', 0, ' ', 0,
-    '0', 0, 'x', 0, '0', 0, '8', 0, '0', 0, '0', 0, '5', 0, '0', 0, '0', 0,
+    '0', 0, 'x', 0, '0', 0, '8', 0, '0', 0, '0', 0, '2', 0, '0', 0, '0', 0,
     '0', 0
 };
 
